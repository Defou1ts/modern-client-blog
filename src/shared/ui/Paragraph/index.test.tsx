import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import { Paragraph } from './index';

const textMock = 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nisi, natus!';

describe('P component', () => {
	it('renders a medium P', () => {
		render(<Paragraph type="medium">{textMock}</Paragraph>);
		const text = screen.getByText(textMock);
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a small P', () => {
		render(<Paragraph type="small">{textMock}</Paragraph>);
		const text = screen.getByText(textMock);
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a label P', () => {
		render(<Paragraph type="label">{textMock}</Paragraph>);
		const text = screen.getByText(textMock);
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('P medium snapshot', () => {
		const component = renderer.create(<Paragraph type="medium">{textMock}</Paragraph>);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('P small snapshot', () => {
		const component = renderer.create(<Paragraph type="small">{textMock}</Paragraph>);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('P label snapshot', () => {
		const component = renderer.create(<Paragraph type="label">{textMock}</Paragraph>);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
