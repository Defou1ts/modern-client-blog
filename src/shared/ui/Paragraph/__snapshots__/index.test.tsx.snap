// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`P component P label snapshot 1`] = `
<p
  className="p label"
>
  Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nisi, natus!
</p>
`;

exports[`P component P medium snapshot 1`] = `
<p
  className="p medium"
>
  Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nisi, natus!
</p>
`;

exports[`P component P small snapshot 1`] = `
<p
  className="p small"
>
  Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nisi, natus!
</p>
`;
